### ISSUE TITLE
Plex Media Player remote control over local network is very unreliable

### ISSUE THREAD

**Issue by twpol** [plexinc/plex-media-player#167]
When using the Plex app on Windows Phone to remotely control Plex Media Player on Windows 10 over a local Wi-Fi network, the ability to pause/resume playback is very unreliable. It is especially so just after unlocking the phone and/or switching back to the Plex app, but this problem is unique to Plex Media Player - it doesn't happen with Plex Home Theater.

The Plex app reports most of the failed attempts as an HTTP status 406 error and waiting a few seconds for the error to disappear usually results in at least one request succeeding. There is no CPU or disk load on the computer running Plex Media Player.
- Windows 10 version 1511 (build 10586.11)
- Plex Media Player for Windows 64bit 1.0.2.70-15a01542
- Plex for Windows Phone 2.1.2.14

I'm guessing this will need logs; phone or PMP or both?

**Comment by PombeirP** [comment]
I wasn't able to repro this with Windows Phone, so it would be good to have the WP logs (and ideally also the PMP ones). What I did notice was that once the phone screen turned off, it could no longer do the periodic timeline subscription, so it ends up unsubscribing:

```
2015-11-30 11:02:18:9983 [46] DEBUG - [PlexHttpClient] - GET http://192.168.16.41:32433/player/timeline/subscribe?protocol=http&port=32467&commandID=720273431
2015-11-30 11:02:19:0608 [23] ERROR - [PlayerSubscriptionManager] - PlexException subscribing to player timeline: Connection to host 192.168.16.41 failed
2015-11-30 11:02:19:0608 [23] WARNING - [PlayerSubscriptionManager] - Unsubscribing from player because subscribe failed with PlexException: Connection to host 192.168.16.41 failed
2015-11-30 11:02:19:0765 [39] DEBUG - [PlexHttpClient] - GET http://192.168.16.41:32433/player/timeline/unsubscribe?commandID=720273432
2015-11-30 11:02:19:0921 [20] ERROR - [PlayerSubscriptionManager] - PlexException unsubscribing from player: Connection to host 192.168.16.41 failed
2015-11-30 11:02:21:3699 [12] DEBUG - [HttpListener] - POST request from 192.168.16.41 for /:/timeline
2015-11-30 11:02:21:3699 [12] DEBUG - [HttpTimelineRequestHandler] - Received POST timeline request from 192.168.16.41
2015-11-30 11:02:21:4308 [12] ERROR - [HttpTimelineRequestHandler] - Timeline request for unsubscribed server: u73f2a3ilrgm0a4i
2015-11-30 11:02:21:4318 [12] DEBUG - [HttpListener] - Response 404 (NotFound) from HttpTimelineRequestHandler with content-length 0
```

**Comment by PombeirP** [comment]
OK, saw the 406 error for the first time just after starting playback remotely on PMP:

it seems that the WP app is not refreshing the subscription when the screen is turned off, and when it tries to communicate again with PMP, the subscription has already been removed:

```
2015-11-30 13:00:13 [ DEBUG ] HTTPServer.cpp @ 146 - Incoming request to: "/player/timeline/subscribe?protocol=http&port=32467&commandID=720276920" from "192.168.16.30" 
2015-11-30 13:00:13 [ DEBUG ] RemoteComponent.cpp @ 389 - Refreshed subscriber: "48f0ee53-c395-4552-9ee6-4a380376d535" 
2015-11-30 13:00:13 [ INFO  ] JS: timeline: playing, 220160/3591840 
2015-11-30 13:00:23 [ INFO  ] JS: timeline: playing, 230240/3591840 
2015-11-30 13:00:33 [ INFO  ] JS: timeline: playing, 240040/3591840 
2015-11-30 13:00:43 [ INFO  ] JS: timeline: playing, 250120/3591840 
2015-11-30 13:00:53 [ INFO  ] JS: timeline: playing, 260200/3591840 
2015-11-30 13:01:03 [ INFO  ] JS: timeline: playing, 270000/3591840 
2015-11-30 13:01:13 [ INFO  ] JS: timeline: playing, 280080/3591840 
2015-11-30 13:01:23 [ INFO  ] JS: timeline: playing, 290160/3591840 
2015-11-30 13:01:33 [ INFO  ] JS: timeline: playing, 300240/3591840 
2015-11-30 13:01:43 [ INFO  ] JS: timeline: playing, 310040/3591840 
2015-11-30 13:01:45 [ DEBUG ] HTTPServer.cpp @ 146 - Incoming request to: "/player/playback/pause?commandID=720276921&type=video" from "192.168.16.30" 
2015-11-30 13:01:45 [ DEBUG ] PlayerComponent.cpp @ 408 - cplayer: Run command: set, flags=0, args=[pause, yes] 
2015-11-30 13:01:45 [ INFO  ] JS: Cannot focus non focusable element 
2015-11-30 13:01:45 [ INFO  ] JS: timeline: paused, 312000/3591840 
2015-11-30 13:01:47 [ DEBUG ] RemoteComponent.cpp @ 457 - more than 10 seconds since we heard from: "Windows-Phone" - unsubscribing.. 
2015-11-30 13:01:47 [ DEBUG ] RemoteComponent.cpp @ 515 - Last subscriber removed, unsubscribing from web 
2015-11-30 13:01:47 [ DEBUG ] RemoteComponent.cpp @ 518 - Removed subscriber: "48f0ee53-c395-4552-9ee6-4a380376d535" 
2015-11-30 13:01:47 [ INFO  ] JS: stopping companion timeline 
2015-11-30 13:01:47 [ WARN  ] RemoteComponent.cpp @ 329 - Could not find responseId: 30  - maybe it was removed because of a timeout? 
2015-11-30 13:01:53 [ INFO  ] JS: timeline: paused, 312000/3591840 
2015-11-30 13:02:03 [ INFO  ] JS: timeline: paused, 312000/3591840 
2015-11-30 13:02:13 [ INFO  ] JS: timeline: paused, 312000/3591840 
2015-11-30 13:02:17 [ DEBUG ] HTTPServer.cpp @ 146 - Incoming request to: "/player/mirror/details?key=%2Flibrary%2Fmetadata%2F9165&machineIdentifier=6be519afe2ae87cfa4ecb67b4cd1a7ca5ab8ad08&protocol=https&port=32400&address=192-168-16-3.75f8e6c9272e4d3c8fed349c493a2e41.plex.direct&token=transient-935e2330-993c-4338-a7f7-d434d11f96bb&commandID=720276922" from "192.168.16.30" 
2015-11-30 13:02:17 [ WARN  ] RemoteComponent.cpp @ 268 - Failed to lock up subscriber "48f0ee53-c395-4552-9ee6-4a380376d535" 
2015-11-30 13:02:22 [ DEBUG ] HTTPServer.cpp @ 146 - Incoming request to: "/player/mirror/details?key=%2Flibrary%2Fmetadata%2F9176&machineIdentifier=6be519afe2ae87cfa4ecb67b4cd1a7ca5ab8ad08&protocol=https&port=32400&address=192-168-16-3.75f8e6c9272e4d3c8fed349c493a2e41.plex.direct&token=transient-2e8aa4a2-8d0e-4a63-a257-846d9126a20e&commandID=720276923" from "192.168.16.30" 
2015-11-30 13:02:22 [ WARN  ] RemoteComponent.cpp @ 268 - Failed to lock up subscriber "48f0ee53-c395-4552-9ee6-4a380376d535" 
2015-11-30 13:02:23 [ INFO  ] JS: timeline: paused, 312000/3591840 
2015-11-30 13:02:24 [ DEBUG ] HTTPServer.cpp @ 146 - Incoming request to: "/player/playback/playMedia?key=%2Flibrary%2Fmetadata%2F9176&mediaIndex=0&containerKey=%2FplayQueues%2F1532&machineIdentifier=6be519afe2ae87cfa4ecb67b4cd1a7ca5ab8ad08&protocol=https&port=32400&address=192-168-16-3.75f8e6c9272e4d3c8fed349c493a2e41.plex.direct&token=transient-d550c112-7a51-4df2-9290-91fb382aad9e&commandID=720276924" from "192.168.16.30" 
2015-11-30 13:02:24 [ WARN  ] RemoteComponent.cpp @ 268 - Failed to lock up subscriber "48f0ee53-c395-4552-9ee6-4a380376d535" 
2015-11-30 13:02:24 [ DEBUG ] HTTPServer.cpp @ 146 - Incoming request to: "/player/timeline/unsubscribe?commandID=720276925" from "192.168.16.30" 
2015-11-30 13:02:24 [ ERROR ] RemoteComponent.cpp @ 504 - Can't remove client: "48f0ee53-c395-4552-9ee6-4a380376d535" since we don't know about it. 
```

We probably need to resubscribe from the WP client after the phone wakes up again, in case the subscription was pruned from a player.

**Comment by PombeirP** [comment]
This may be fixed with recent changes in WP 2.1.2.23.
